src/global/reco/InclusiveKinematicsDA_factory.h

```diff
@@ -13,36 +13,38 @@
 #include <vector>
 
 #include "algorithms/reco/InclusiveKinematicsDA.h"
-#include "extensions/jana/JChainMultifactoryT.h"
+#include "extensions/jana/JOmniFactory.h"
 #include "extensions/spdlog/SpdlogMixin.h"
 
 namespace eicrecon {
 
-    class InclusiveKinematicsDA_factory :
-            public JChainMultifactoryT<>,
-            public SpdlogMixin {
-
-    public:
-
-        explicit InclusiveKinematicsDA_factory(
-            std::string tag,
-            const std::vector<std::string>& input_tags,
-            const std::vector<std::string>& output_tags)
-        : JChainMultifactoryT<>(std::move(tag), input_tags, output_tags) {
-
-            DeclarePodioOutput<edm4eic::InclusiveKinematics>(GetOutputTags()[0]);
-
-        }
-
-        /** One time initialization **/
-        void Init() override;
-
-        /** Event by event processing **/
-        void Process(const std::shared_ptr<const JEvent> &event) override;
-
-    protected:
-        InclusiveKinematicsDA m_inclusive_kinematics_algo;
-
-    };
+class InclusiveKinematicsDA_factory :
+        public JOmniFactory<InclusiveKinematicsDA_factory> {
+
+private:
+    using AlgoT = eicrecon::InclusiveKinematicsDA;
+    std::unique_ptr<AlgoT> m_algo;
```

Comment:
@nathanwbrei The reason for this unique_ptr pattern is to allow instantiation of the algorithm with name set to `GetPrefix()` as required by the eic/algorithms interface. That interface allows setting name only once, on in instantiation.